tt_eager/tt_dnn/op_library/moreh_sum_backward/moreh_sum_backward_op.cpp

```diff
@@ -100,18 +106,24 @@ operation::ProgramWithCallbacks MorehSumBackward::create_program(
 
 Tensor moreh_sum_backward(
     const Tensor &output_grad,
-    const Tensor &input,
+    const std::optional<const Tensor> input,
     std::optional<std::variant<int64_t, std::vector<int64_t>>> dim,
     const bool keep_batch_dim,
     const std::optional<const Tensor> input_grad,
     const MemoryConfig &input_grad_mem_config,
     std::optional<const DeviceComputeKernelConfig> compute_kernel_config) {
 
-    std::vector<int64_t> dims = get_dim(dim, input.get_legacy_shape().rank());
+    TT_FATAL((input.has_value() || input_grad.has_value()), "either input or input_grad must have a value");
```

Comment:
The message needs to be updated because `input_grad` is always required, while `input` is only necessary when implementing `ignore_nan`.